numpy/lib/function_base.py

```diff
@@ -4017,8 +4017,10 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
             "'midpoint', or 'nearest'")
 
     # The dimensions of `q` are prepended to the output shape, so we need the
-    # axis being sampled from `ap` to be first.
-    ap = np.moveaxis(ap, axis, 0)
+    # axis being sampled from `ap` to be first.  But we only do so if axis!=0,
+    # as moveaxis is quite slow.
```

Comment:
Can we put this optimization in `moveaxis` if its really that slow?

Reply:
that's slightly annoying because you'd have to write something like `if np.all(source == destination)` (as the arguments can be either scalars and vectors) but then you're paying a full array creation and iteration overhead even if source and destination were scalars to start with...
IOW the difference here is that we know the arguments are scalars.

Reply:
I was thinking a naive approach like `if source is destination` inside moveaxis would be enough to speed this up in many cases, since the small integers are cached.

Reply:
sure.